### PR TITLE
Now Send/Request/ResponseFn accept transformers

### LINKED ISSUE #451: Accept transformers is Send/Request method signatures
Because transformers are usually per-request, it doesn't make sense to assign them to a Send/Request implementation like here: https://github.com/cloudevents/sdk-go/blob/master/v2/protocol/http/protocol.go#L27

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fixes #451

Also removes `transformer.SetUUID`

Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

### CODE REVIEW

File: v2/protocol/http/protocol.go [hunk 1]
```diff
@@ -24,7 +24,6 @@ const (
 type Protocol struct {
 	Target          *url.URL
 	RequestTemplate *http.Request
-	transformers    binding.Transformers
```
Comment (n3wscott) [inline v2/protocol/http/protocol.go:27]: don't you want some static ones set? or they should always come from the outside?

**Comment by slinkydeveloper** [comment]
@n3wscott they are not meant to be usable from client. If the user wants to perform a trasformation at client level, why bother using transformers? He can directly operate on the `Event`

**Comment by n3wscott** [comment]
Well look how time or Id defaulting works.

**Comment by slinkydeveloper** [comment]
> Well look how time or Id defaulting works.

The client uses the defaulters in `client/defaulters.go`.

I think the UUID defaulter as transformer should be removed frankly

**Comment by slinkydeveloper** [comment]
I mean, the client impl could always use the transformers, not the user directly

**Pull request merged**